src/translate.py

```diff
@@ -431,29 +446,78 @@ def get_stack_info(
         ):
             # Saving the return address on the stack.
             info.is_leaf = False
-            info.return_addr_location = inst.args[1].lhs_as_literal()
+            stack_offset = inst.args[1].lhs_as_literal()
+            info.return_addr_location = stack_offset
+            callee_saved_words.append(stack_offset)
         elif (
             inst.mnemonic in ["sw", "swc1", "sdc1"]
             and destination in SAVED_REGS
             and isinstance(inst.args[1], AsmAddressMode)
             and inst.args[1].rhs.register_name == "sp"
+            and destination not in info.callee_save_reg_locations
         ):
             # Initial saving of callee-save register onto the stack.
-            info.callee_save_reg_locations[destination] = inst.args[1].lhs_as_literal()
-
-    # Find the region that contains local variables. It is above saved registers
-    # and the return address, if those exist. If they don't, the local variables
-    # can lie directly at the bottom of the stack.
-    info.local_vars_region_bottom = 0
+            stack_offset = inst.args[1].lhs_as_literal()
+            info.callee_save_reg_locations[destination] = stack_offset
+            if inst.mnemonic == "sdc1":
+                callee_saved_doubles.append(stack_offset)
+            else:
+                callee_saved_words.append(stack_offset)
+        elif (
+            inst.mnemonic in ["lw", "lwc1", "ldc1"]
+            and isinstance(inst.args[1], AsmAddressMode)
+            and inst.args[1].rhs.register_name == "sp"
+        ):
```

Comment:
the surrounding loop is only over the first basic block, but we may want to run this on the entire function

Reply:
Great point - running over the whole function fixes the `ovl_En_Go_0x80A10FD0` weirdness I commented about.